### PR TITLE
Replaced BigNumber with BigInt

### PR DESCRIPTION
**Pull request by popenta**
For the new v13 release, the functionality added will use `bigint` instead of `BigNumber`.

### CODE REVIEW

File: src/transactionsFactories/tokenManagementTransactionsFactory.ts [hunk 1]
```diff
@@ -575,14 +583,14 @@ export class TokenManagementTransactionsFactory {
...
+        tokenNonce: bigint;
+        quantityToBurn: bigint;
     }): TransactionNext {
         const dataParts = [
             "ESDTNFTBurn",
             utf8ToHex(options.tokenIdentifier),
-            bigIntToHex(options.tokenNonce),
-            bigIntToHex(options.quantityToBurn),
+            bigIntToHex(new BigNumber(options.tokenNonce.toString())),
+            bigIntToHex(new BigNumber(options.quantityToBurn.toString())),
```
Comment (andreibancioiu) [inline src/transactionsFactories/tokenManagementTransactionsFactory.ts:593]: Perhaps use `numberToPaddedHex` (which handles `bigint`). If necessary, maybe handle 0 differently in `numberToPaddedHex`.

File: src/transaction.spec.ts [hunk 2]
```diff
@@ -32,8 +32,8 @@ describe("test transaction construction", async () => {
         const transaction = Transaction.fromTransactionNext(nextTransaction);
         assert.deepEqual(transaction.getSender(), Address.fromBech32(plainTransactionNextObject.sender));
         assert.deepEqual(transaction.getReceiver(), Address.fromBech32(plainTransactionNextObject.receiver));
-        assert.equal(transaction.getGasLimit().valueOf(), plainTransactionNextObject.gasLimit);
-        assert.equal(transaction.getValue().toString(), plainTransactionNextObject.value);
+        assert.equal(transaction.getGasLimit().valueOf().toFixed(0), plainTransactionNextObject.gasLimit.toString());
```
[the remark this answers is not in the document]
Comment (popenta) [reply src/transaction.spec.ts:35]: `toString()` works fine as well.

File: src/transaction.ts [hunk 3]
```diff
@@ -428,9 +428,9 @@ export class Transaction {
     const tx = new Transaction({
       sender: Address.fromBech32(transaction.sender),
       receiver: Address.fromBech32(transaction.receiver),
-      gasLimit: new BigNumber(transaction.gasLimit).toNumber(),
+      gasLimit: new BigNumber(transaction.gasLimit.toString()).toNumber(),
```
[the remark this answers is not in the document]
Comment (popenta) [reply src/transaction.ts:431]: Done

File: src/transactionsFactories/tokenManagementTransactionsFactory.ts (hunk 1 above)
Comment (popenta) [reply src/transactionsFactories/tokenManagementTransactionsFactory.ts:593]: Replaced with `numberToPaddedHex`.

File: src/transactionsFactories/tokenManagementTransactionIntentsFactory.spec.ts [hunk 4]
```diff
@@ -49,7 +49,7 @@ describe("test token management transactions factory", () => {
         assert.deepEqual(
             next.data,
             Buffer.from(
-                "issue@4652414e4b@4652414e4b@64@@63616e467265657a65@74727565@63616e57697065@74727565@63616e5061757365@74727565@63616e4368616e67654f776e6572@74727565@63616e55706772616465@66616c7365@63616e4164645370656369616c526f6c6573@66616c7365",
+                "issue@4652414e4b@4652414e4b@64@00@63616e467265657a65@74727565@63616e57697065@74727565@63616e5061757365@74727565@63616e4368616e67654f776e6572@74727565@63616e55706772616465@66616c7365@63616e4164645370656369616c526f6c6573@66616c7365",
```
Comment (andreibancioiu) [inline src/transactionsFactories/tokenManagementTransactionIntentsFactory.spec.ts:52]: Should be fine :+1:

**Review by andreibancioiu — APPROVED** [review]

File: src/transaction.ts [hunk 5]
```diff
@@ -614,26 +614,24 @@ export class TransactionNext {
...
     if (moveBalanceGas === transaction.gasLimit) {
       return feeForMove;
     }
 
-    const diff = new BigNumber(transaction.gasLimit).minus(moveBalanceGas);
-    const modifiedGasPrice = gasPrice.multipliedBy(
-      new BigNumber(networkConfig.GasPriceModifier)
-    );
-    const processingFee = diff.multipliedBy(modifiedGasPrice);
+    const diff = transaction.gasLimit- moveBalanceGas;
```
Comment (bogdan-rosianu) [inline src/transaction.ts:630]: `transaction.gasLimit- moveBalanceGas;`
=>
`transaction.gasLimit - moveBalanceGas;`

File: src/transactionsFactories/tokenManagementTransactionIntentsFactory.spec.ts (hunk 4 above)
Comment (bogdan-rosianu) [reply src/transactionsFactories/tokenManagementTransactionIntentsFactory.spec.ts:52]: confirmed on testnet 👍

File: src/transaction.ts (hunk 5 above)
Comment (popenta) [reply src/transaction.ts:630]: Done

**Review by bogdan-rosianu — APPROVED** [review]

**Review by andreibancioiu — APPROVED** [review]

**Pull request merged**